node/consensus/data/consensus_frames.go

```diff
@@ -24,30 +24,24 @@ func (e *DataClockConsensusEngine) collect(
 	e.logger.Info("collecting vdf proofs")
 
 	latest := enqueuedFrame
-
 	for {
-		peerId, maxFrame, err := e.GetMostAheadPeer(e.latestFrameReceived)
```

Comment:
Getting the most ahead peer based only on `e.latestFrameReceived` is actually wrong at times - `latest` may be ahead of `e.latestFrameReceived` now, and we don't need to sync actually.